atsign-foundation/noports
# feat: NoPorts policy plane aka delegated authorization

**- What I did**
- Initial implementation of a NoPorts policy plane aka delegated authorization.
- Daemons may talk to another atSign (the policy manager) to ask whether a given client atSign may connect to it.
- closes https://github.com/atsign-foundation/noports/issues/855

**- How I did it**
- New `npa` library in `noports_core` package
- New demo binaries in `sshnoports` package
- New command-line option for sshnpd : `-p` or `--policy-manager`

**- How to verify it**
- Automated tests should pass
- Manual tests
  - cd /path/to/repo/packages/dart/sshnoports
  - Run `./buildBinaries`
  - Run the demo NoPorts policy CLI program, a NoPorts daemon and a NoPorts client like this:
    ```
    bin/demo/npa_cli -a "@policy_atsign" --daemon-atsigns "@daemon_atsign[,...]"
    bin/sshnpd -p @policy_atsign -a @daemon_atsign -d devicename -u -s -v
    bin/sshnp -f @client_atsign -d devicename -s ~/.ssh/IDFILE.pub -i ~/.ssh/IDFILE -t @daemon_atsign -h @rv_am
    ```
